cmd/ctl/pkg/status/certificate/certificate.go

```diff
@@ -0,0 +1,169 @@
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	restclient "k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+Get details about the current status of a Certificate, including information on related resources like CertificateRequest or Order.`))
+
+	example = templates.Examples(i18n.T(`
+# Query status of cert-manager Certificate resource with name my-cert in namespace default
+kubectl cert-manager status certificate my-cert --namespace default
+`))
+)
+
+// Options is a struct to support status certificate command
+type Options struct {
+	CMClient   cmclient.Interface
+	RESTConfig *restclient.Config
+	// The Namespace that the Certificate to be renewed resided in.
+	// This flag registration is handled by cmdutil.Factory
+	Namespace string
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdStatusCert returns a cobra command for status certificate
+func NewCmdStatusCert(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:     "certificate",
+		Short:   "Get details about the current status of a Certificate",
+		Long:    long,
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Validate(args))
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(args []string) error {
+	if len(args) < 1 {
+		return errors.New("the name of the Certificate has to be provided as argument")
+	}
+	if len(args) > 1 {
+		return errors.New("only one argument can be passed in: the name of the Certificate")
+	}
+	return nil
+}
+
+// Complete takes the factory and infers any remaining options.
+func (o *Options) Complete(f cmdutil.Factory) error {
+	var err error
+
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig, err = f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.CMClient, err = cmclient.NewForConfig(o.RESTConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes status certificate command
+func (o *Options) Run(args []string) error {
+	ctx := context.TODO()
+	crtName := args[0]
+
+	crt, err := o.CMClient.CertmanagerV1alpha2().Certificates(o.Namespace).Get(ctx, crtName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error when getting Certificate resource: %v", err)
+	}
+
+	fmt.Fprintf(o.Out, fmt.Sprintf("Name: %s\nNamespace: %s\n", crt.Name, crt.Namespace))
```

Comment:
This looks good for now - I wonder if in future we should have a function like `printObjectMeta(runtime.Object, io.Writer)` which will print the Name and Namespace for an object, perhaps with a `prefix string` argument too which can be used to prefix a string to each line (e.g. if we want to have the object meta printed but indented by two spaces, so this can be re-used for printing CertificateRequest metadata etc in future too)